### ISSUE TITLE
DAB+ from TRT Turkey

### ISSUE THREAD

**Issue by jpuigs** [Opendigitalradio/dablin#42]
TRT DAB+ signal from Turkey

$ curl http://...........  | dablin_gtk.exe

DABlin v1.10.0 - capital DAB experience
Plays a DAB/DAB+ audio service from a frame-aligned ETI-NI stream.
  https://github.com/Opendigitalradio/dablin
SDLOutput: using SDL version '2.0.7'
ETISource: reading from 'stdin'
ETIPlayer: ignored ETI frame with FSYNC = 0x6F1910
ETIPlayer: ignored ETI frame with FSYNC = 0xDC44A5
ETIPlayer: ignored ETI frame with FSYNC = 0xFFE694
ETIPlayer: ignored ETI frame with FSYNC = 0xDA198B
ETIPlayer: ignored ETI frame with FSYNC = 0xA70C59
ETIPlayer: ignored ETI frame with FSYNC = 0x422014
ETIPlayer: ignored ETI frame with FSYNC = 0xF23DE2
ETIPlayer: ignored ETI frame with FSYNC = 0xE1E886
ETIPlayer: ignored ETI frame with FSYNC = 0x031C6D
ETIPlayer: ignored ETI frame with FSYNC = 0x7C02E2
ETIPlayer: ignored ETI frame with FSYNC = 0x560000
……..

and it does not play audio.
Since 3 days ago.

**Comment by jpuigs** [comment]
[trt.zip](https://github.com/Opendigitalradio/dablin/files/2722377/trt.zip)

**Comment by andimik** [comment]
Does it work with `ni2http - - list` ?

We already had that. You have to patch the file and crop the first x bytes.

**Comment by andimik** [comment]
See https://github.com/piratfm/eti-tools/issues/28#issuecomment-427292468

**Comment by andimik** [comment]
Then it was frame aligned by chance.

**Comment by basicmaster** [comment]
Indeed here the reason is the one that @andimik refers to - DABlin is designed to play frame-aligned ETI. In the case of your testfile, `dd skip=1 bs=5288 < trt.eti | dablin_gtk` will skip the bytes until a new ETI frame begins.

From which source/server do you retrieve the ETI stream? Maybe that software can be modified to only start streaming on ETI frame boundaries.

I haven't had a look into the raw stream of TS packets of such an ETI transmission yet, but would guess that the software starts output only at specific locations (Payload Unit Start Indicator). So it would make more sense to only check for the start of an ETI frame at that locations in the server, instead of at every byte in DABlin itself...

**Comment by jpuigs** [comment]
I get the ETI stream from a live source, an enigma 2 satellite receiver.
$ curl http://192.168.0/........... | dablin_gtk
So, I really don't know how can I make skip those bytes in a live stream.....

Using welle.io this test file, and Live stream both work fine, but in this case I have to convert to raw.
$ curl http://192.168.0 | odr-dabmod eti2raw.ini | ./welle.io_1.0_Linux_x86_64.AppImage --raw-file /dev/stdin -d rawfile

**Comment by andimik** [comment]
Yes, you are doing exactly the same like me.

**Comment by andimik** [comment]
BTW
The TRT DAB stream seems to have got new content compared to my previous check (can't receive 42 East) and https://www.fmlist.org/sendertabelle/dab-ww_more.php?inputLand=TUR&inputEn=4FFF

**Comment by andimik** [comment]
I use the Greek fork of eti-tools https://github.com/satdreamgr/eti-tools

**Comment by jpuigs** [comment]
> So the question is, what software/tool is running on your enigma STB which serves the stream that you access with `curl`. Is it `ts2na` with some HTTP server?

Yes, this turkish stream is the output of a ts2na.

**Comment by andimik** [comment]
What happens if you curl the TS and process the rest on your PC?

That's what I do.

But I don't receive 42 East with my dish. I only have recorded TS.

**Comment by jpuigs** [comment]
It does exactly the same:

$ wget http://192.168.0.16:8001/1:0:64:1:464E:0:1A4AAC9:0:0:0: -q -O- | ./ts2na -p 1068 -s 0 | dablin_gtk 
INFO:  Using pid: 0x042c (1068)
DABlin v1.10.0 - capital DAB experience
Plays a DAB/DAB+ audio service from a frame-aligned ETI-NI stream.
  https://github.com/Opendigitalradio/dablin

SDLOutput: using SDL version '2.0.8'
ETISource: reading from 'stdin'
ETIPlayer: ignored ETI frame with FSYNC = 0x555555
ETIPlayer: ignored ETI frame with FSYNC = 0x555555
ETIPlayer: ignored ETI frame with FSYNC = 0x555555
ETIPlayer: ignored ETI frame with FSYNC = 0x555555
ETIPlayer: ignored ETI frame with FSYNC = 0x555555
ETIPlayer: ignored ETI frame with FSYNC = 0x555555
....

**Comment by basicmaster** [comment]
Could you provide a dump of 1-2 minutes of the MPEG-TS which contains the stream - but only of the affected PID 1068? This would allow to check for possible regularities that enable a fast sync.

**Comment by andimik** [comment]
Here you are (version from October)

```
$ dvbstream -analyse -stdin < trt_okt2018_short.ts 
dvbstream v0.7 - (C) Dave Chapman 2001-2004
Released under the GPL.
Latest version available from http://www.linuxstb.org/
dvbstream will stop after 10 seconds (0 minutes)
Analysing PIDS
Caught signal 1 - closing cleanly.
1068,15.660 Mbit/s
```
[trt_okt2018_short.zip](https://github.com/Opendigitalradio/dablin/files/2727866/trt_okt2018_short.zip)

**Comment by jpuigs** [comment]
... and isn't easier to correct it inisde dablin ?
ni2out (ni2http) gives sound and listings, so do they correct it ?

**Comment by basicmaster** [comment]
> ... and isn't easier to correct it inisde dablin ?

Initially I thought there would be a conflict between a sync algorithm (to process non-frame-aligned ETI) and the progress bar (which relies on the total file size and the resulting total frames count; in case the file size is known). So that a separate tool was required to prevent the progress feature to be removed.

But when I wrote such tool yesterday, it turned out that it would not be that hard to integrate the sync algorithm into DABlin itself - while keeping the progress calculation, or rather to say, to extend the progress calculation to adjust the progress on discarded data.

So I added a sync algorithm to DABlin that discards any data before/after ETI-NI frames. This solves the original issue.
